docs/references/amp-constants.md

```diff
@@ -0,0 +1,26 @@
+# AMP Constants
+
+Amp Constants are the information passed from the AMP Engine to the tool
+
+## Constants
+Here is a list of constants passed from the engine to the tool:
+- **_designer_version** the current version of designer
+- **_alteryx_install_dir** the path to the installation directory of the designer executable
+- **_workflow_dir** the path to the directory the current workflow is in
+- **_temp_dir** the path to the temp directory where runtime data is stored
+- **_tool_id** the id of the tool
+- **_workflow_id** the id of the current workflow
+- **_raw_constants** the rest of the data sent by the engine in it's raw format, currently this consists of two pieces of information:
+    - **Engine.WorkflowFileName** the file name of the workflow
+    - **Engine.Type** the type of engine
```

Comment:
Also, there's more stuff in the `raw_constants` dict - I suggest dumping the dict with `provider.io` to get a full accounting. Something like:

```
for k, v in self.provider.environment.raw_constants.sorted().items():
    self.provider.io.info(f"{k}: {v}")
```

should work in Designer